test/fake_slack.ml

```diff
@@ -0,0 +1,121 @@
+(* A (currently pretty useless) fake slack implementation to run tests
+   against. *)
+open Lwt
+open Cohttp
+open Cohttp_lwt_unix
+
+
+let valid_token = "xoxp-testtoken" (* The only "valid" token we accept. *)
+
+(* The following values come from data captured by a relay proxy between slacko
+   and slack.com. They need to be replaced with something more generic that
+   doesn't depend on the arbitrary details of a particular slack team. *)
+let ch_general = "C3UK9TS3C"
+let ch_random = "C3TTWNCTA"
+let ch_archivable = "C3XTJPLFL"
+let ch_archived = "C3XTHDCTC"
+
+
+let channels_json = Yojson.Safe.from_file "channels.json"
+let new_channel_json = Yojson.Safe.from_file "new_channel.json"
+let authed_json = Yojson.Safe.from_file "authed.json"
+let random_history_json = Yojson.Safe.from_file "random_history.json"
+let users_json = Yojson.Safe.from_file "users.json"
```

Comment:
There isn't really enough commonality between the different endpoints to make that work easily. For some of them (such as the history APIs), we'll also need some more complex logic to decide what to return.

I'm hoping to eventually replace this with a more generic test double that doesn't rely on a bunch of stored data, but that's too big for this PR.